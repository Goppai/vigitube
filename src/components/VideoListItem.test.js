import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { VideoListItem } from './'

describe('<VideoListItem/>', () => {
  const onClick = jest.fn()
  const dummyItem = {
    snippet: {
      thumbnails: {
        default: {
          url: 'some-url'
        }
      },
      title: 'Some title'
    }
  }
  const wrapper = shallow(<VideoListItem onClick={onClick} item={dummyItem}/>)

  it('renders', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('handles click', () => {
    wrapper.simulate('click')
    expect(onClick).toHaveBeenCalled()
  })
})
