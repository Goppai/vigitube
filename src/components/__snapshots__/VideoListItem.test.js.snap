// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoListItem/> renders 1`] = `
<Item
  className="video-list-item"
  extra={
    <img
      alt="video-thumbnail"
      src="some-url"
      width={100}
    />
  }
  onClick={[Function]}
>
  <Meta
    title="Some title"
  />
</Item>
`;
